### PR TITLE
lightning: refactor to reuse in load data part 5

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #40499 

Problem Summary:

### What is changed and how it works?
- decouple backend and engine manage code(particially) to use local backend directly in load data
- remove `noop` backend. remove `LocalEngineWriter` wrapper. 
- remove tableName param from writer
- move some code unique to local backend to itself

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/backend.go [hunk 1]
```diff
@@ -221,106 +202,17 @@ type OpenedEngine struct {
...
-// // import_ the data written to the engine into the target.
-// import_(ctx context.Context) error
-
-// // cleanup deletes the imported data.
-// cleanup(ctx context.Context) error
-
-// ClosedEngine represents a closed engine, allowing ingestion into the target.
-// This type is goroutine safe: you can share an instance and execute any method
-// anywhere.
-type ClosedEngine struct {
```
Comment (D3Hunter) [inline br/pkg/lightning/backend/backend.go:233]: moved to below where its methods are defined

**Comment by D3Hunter** [comment]
ci pass https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_br_test/detail/tidb_ghpr_integration_br_test/12516/pipeline

File: br/pkg/lightning/backend/local/disk_quota.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+type DiskUsage interface {
+	// EngineFileSizes obtains the size occupied locally of all engines managed
+	// by this backend. This method is used to compute disk quota.
+	// It can return nil if the content are all stored remotely.
+	EngineFileSizes() (res []backend.EngineFileSize)
+}
+
+// CheckDiskQuota verifies if the total engine file size is below the given
+// quota. If the quota is exceeded, this method returns an array of engines,
+// which after importing can decrease the total size below quota.
```
Comment (lance6716) [inline br/pkg/lightning/backend/local/disk_quota.go:33]: How can we use `inProgressLargeEngines`?

**Review by lance6716 — COMMENTED** [review]
14 / 34 files viewed

**Review by lance6716 — APPROVED** [review]

File: br/pkg/lightning/backend/local/disk_quota.go (hunk 2 above)
Comment (D3Hunter) [reply br/pkg/lightning/backend/local/disk_quota.go:33]: didn't look in detail, just copied out

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**